client/navigation/components/Item/index.js

```diff
@@ -2,17 +2,30 @@
  * External dependencies
  */
 import { __experimentalNavigationItem as NavigationItem } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
 import { WooNavigationItem, useNavSlot } from '@woocommerce/navigation';
 
 const Item = ( { item } ) => {
 	const slot = useNavSlot( 'woocommerce_navigation_' + item.id );
 	const hasFills = Boolean( slot.fills && slot.fills.length );
 
+	const trackClick = ( id ) => {
+		recordEvent( 'navigation_click', {
+			menu_item: id,
+		} );
+	};
+
+	// Disable reason: The div wrapping the slot item is used for tracking purposes
+	// and should not be a tabbable element.
+	/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
+
 	// Only render a slot if a coresponding Fill exists and the item is not a category
 	if ( hasFills && ! item.isCategory ) {
 		return (
 			<NavigationItem key={ item.id } item={ item.id }>
-				<WooNavigationItem.Slot name={ item.id } />
+				<div onClick={ () => trackClick( item.id ) }>
```

Comment:
Does this `onClick` not work directly on `NavigationItem`?

Reply:
@psealock It currently does not; see https://github.com/WordPress/gutenberg/blob/47feef0a2aaaa38b34fb3525a35cfddaf032c719/packages/components/src/navigation/item/index.js#L73

The children replaces the `ItemUI` which is what houses the onclick props.  We could potentially open a PR in GB to try and pass these props to `children` as well, but I think this would still require the child element to accept and use the `onClick` prop in its component.